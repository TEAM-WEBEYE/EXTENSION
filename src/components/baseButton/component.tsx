import React from "react";

interface BaseButtonProps {
    children: React.ReactNode;
    onClick: () => void;
    ariaLabel?: string;
    color?: "gray" | "white" | "purple" | "black" | "yellow" | "dark";
    isSelected?: boolean;
}

const buttonStyles = {
    gray: "bg-grayscale-200 text-grayscale-900 border-grayscale-300 hover:bg-grayscale-200 hover:border-grayscale-300 hover:text-grayscale-400 ",
    white: "bg-grayscale-100 text-grayscale-900 border-grayscale-300 hover:bg-grayscale-100 hover:border-grayscale-300 hover:text-grayscale-500",
    purple: "bg-purple-default text-grayscale-100 border-none hover:bg-purple-light hover:border-none hover:text-grayscale-100",
    yellow: "bg-yellow-default text-grayscale-900 border-none hover:bg-yellow-light hover:border-none hover:text-grayscale-600",
    black: "bg-grayscale-900 text-grayscale-100 border-grayscale-800 hover:bg-grayscale-800 hover:border-grayscale-800 hover:text-grayscale-500",
    dark: "bg-grayscale-800 text-grayscale-100 border-grayscale-800 hover:bg-grayscale-700 hover:border-grayscale-800 hover:text-grayscale-500",
};

const selectedStyles = {
    gray: "bg-grayscale-200 border-purple-default text-grayscale-900",
    white: "bg-grayscale-100 border-purple-default text-grayscale-900",
    purple: "bg-purple-default border-purple-default text-grayscale-100",
    yellow: "bg-yellow-default border-purple-default text-grayscale-900",
    black: "bg-grayscale-900 border-purple-default text-grayscale-100",
    dark: "bg-grayscale-800 border-purple-default text-grayscale-100",
};

const baseClasses =
    "rounded-[14px] w-[140px] h-[140px] flex items-center justify-center border-4 border-solid font-24-Regular font-koddi";

export function BaseButton({
    children,
    onClick,
    ariaLabel,
    color = "gray",
    isSelected = false,
}: BaseButtonProps) {
    const colorClasses = isSelected
        ? selectedStyles[color]
        : buttonStyles[color];

    return (
        <button
            className={`${baseClasses} ${colorClasses}`}
            onClick={onClick}
            aria-label={ariaLabel}
            aria-pressed={isSelected}
        >
            {children}
        </button>
    );
}
