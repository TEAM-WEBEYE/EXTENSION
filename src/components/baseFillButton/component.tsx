import { useTheme } from "@src/contexts/ThemeContext";
import React from "react";

interface BaseFillButtonProps {
    children: React.ReactNode;
    onClick: () => void;
    ariaLabel?: string;
    isDisabled?: boolean;
}

export function BaseFillButton({
    children,
    onClick,
    ariaLabel,
    isDisabled = false,
}: BaseFillButtonProps) {
    const { theme, fontClasses } = useTheme();
    const isDarkMode = theme === "dark";

    return (
        <button
            className={`font-koddi ${fontClasses.fontCommon} py-[16px] px-[30px] relative w-full rounded-[14px]  ${
                isDisabled
                    ? isDarkMode
                        ? "bg-grayscale-400 text-grayscale-100"
                        : " text-grayscale-100  bg-grayscale-600"
                    : isDarkMode
                      ? "bg-purple-light text-grayscale-900 hover:opacity-30 active:bg-purple-dark active:hover:opacity-100"
                      : "text-grayscale-100  bg-purple-default hover:opacity-30 active:bg-purple-dark active:hover:opacity-100"
            }`}
            onClick={onClick}
            aria-label={ariaLabel}
            aria-pressed={isDisabled}
            disabled={isDisabled}
        >
            {children}
        </button>
    );
}
