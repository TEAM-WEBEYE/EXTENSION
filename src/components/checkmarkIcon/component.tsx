import React from "react";
import { useTheme } from "@src/contexts/ThemeContext";

export function CheckmarkIcon(
    props: React.SVGProps<SVGSVGElement>,
): JSX.Element {
    const { theme: themeMode } = useTheme();
    const isDarkMode = themeMode === "dark";

    return (
        <svg
            {...props}
            xmlns="http://www.w3.org/2000/svg"
            width={34}
            height={34}
            viewBox="0 0 34 34"
            fill="none"
        >
            <circle
                cx="17"
                cy="17"
                r="15"
                fill={isDarkMode ? "#323335" : "#fefefe"}
            />
            <path
                d="M14.9 19.7L11.675 16.475C11.4 16.2 11.05 16.0625 10.625 16.0625C10.2 16.0625 9.85 16.2 9.575 16.475C9.3 16.75 9.1625 17.1 9.1625 17.525C9.1625 17.95 9.3 18.3 9.575 18.575L13.85 22.85C14.15 23.15 14.5 23.3 14.9 23.3C15.3 23.3 15.65 23.15 15.95 22.85L24.425 14.375C24.7 14.1 24.8375 13.75 24.8375 13.325C24.8375 12.9 24.7 12.55 24.425 12.275C24.15 12 23.8 11.8625 23.375 11.8625C22.95 11.8625 22.6 12 22.325 12.275L14.9 19.7ZM17 32C14.925 32 12.975 31.606 11.15 30.818C9.325 30.03 7.7375 28.9615 6.3875 27.6125C5.0375 26.2635 3.969 24.676 3.182 22.85C2.395 21.024 2.001 19.074 2 17C1.999 14.926 2.393 12.976 3.182 11.15C3.971 9.324 5.0395 7.7365 6.3875 6.3875C7.7355 5.0385 9.323 3.97 11.15 3.182C12.977 2.394 14.927 2 17 2C19.073 2 21.023 2.394 22.85 3.182C24.677 3.97 26.2645 5.0385 27.6125 6.3875C28.9605 7.7365 30.0295 9.324 30.8195 11.15C31.6095 12.976 32.003 14.926 32 17C31.997 19.074 31.603 21.024 30.818 22.85C30.033 24.676 28.9645 26.2635 27.6125 27.6125C26.2605 28.9615 24.673 30.0305 22.85 30.8195C21.027 31.6085 19.077 32.002 17 32Z"
                fill={isDarkMode ? "#B872FF" : "#8914FF"}
            />
        </svg>
    );
}
