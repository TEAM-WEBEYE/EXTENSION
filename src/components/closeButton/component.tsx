import { useAppTheme } from "@src/contexts/ThemeContext";
import React from "react";

interface CloseButtonProps {
    onClick: () => void;
    ariaLabel?: string;
}

export function CloseButton({ onClick, ariaLabel = "닫기" }: CloseButtonProps) {
    const { theme } = useAppTheme();
    const isDarkMode = theme === "dark";

    const strokeColor = isDarkMode ? "#FEFEFE" : "#121212";

    return (
        <button
            className={` p-[16px] relative rounded-[14px] ${
                isDarkMode
                    ? "bg-grayscale-900  hover:opacity-30 border-4 border-solid border-grayscale-700 active:border-purple-light active:hover:opacity-100"
                    : "bg-grayscale-100 hover:opacity-30 border-4 border-solid border-grayscale-300 active:border-purple-default active:hover:opacity-100"
            }`}
            onClick={onClick}
            aria-label={ariaLabel}
        >
            <svg
                xmlns="http://www.w3.org/2000/svg"
                width="36"
                height="36"
                viewBox="0 0 36 36"
                fill="none"
            >
                <path
                    d="M27 8.99805L9 26.998M27 26.998L9 8.99805"
                    stroke={strokeColor}
                    strokeWidth="3"
                    strokeLinecap="round"
                />
            </svg>
        </button>
    );
}
