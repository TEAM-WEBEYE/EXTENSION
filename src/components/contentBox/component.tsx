import { useAppTheme } from "@src/contexts/ThemeContext";
import React from "react";

interface ContentBoxProps {
    children: React.ReactNode;
    ariaLabel?: string;
}

export function ContentBox({ children, ariaLabel }: ContentBoxProps) {
    const { theme, fontClasses } = useAppTheme();

    const isDarkMode = theme === "dark";
    return (
        <div
            className={`${fontClasses.fontCommon} font-koddi px-[24px] py-[18px] rounded-[14px] flex justify-between gap-[110px]
            ${isDarkMode ? `bg-grayscale-800 text-grayscale-100` : `bg-grayscale-200 text-grayscale-900`}`}
            aria-label={ariaLabel}
        >
            {children}
        </div>
    );
}
