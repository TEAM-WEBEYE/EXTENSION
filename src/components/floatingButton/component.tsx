import React from "react";
import styled from "@emotion/styled";

interface FloatingButtonProps {
    onClick: () => void;
    iconUrl: string;
}

const ButtonContainer = styled.div`
    position: fixed;
    top: 70px;
    right: 20px;
    width: 58px;
    height: 58px;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
    z-index: 9999;
    transition: all 0.2s ease-in-out;
    &:hover {
        transform: scale(1.05);
    }
`;

const ButtonImage = styled.img`
    width: 100%;
    height: 100%;
    max-width: 58px;
    max-height: 58px;
    object-fit: contain;
`;

export function FloatingButton({ onClick, iconUrl }: FloatingButtonProps) {
    console.log("FloatingButton iconUrl:", iconUrl);
    return (
        <ButtonContainer onClick={onClick} className="select-none">
            <ButtonImage src={iconUrl} alt="Floating Button" />
        </ButtonContainer>
    );
}
