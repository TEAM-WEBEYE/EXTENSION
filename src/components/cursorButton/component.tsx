import React from "react";
import { BaseButton } from "../baseButton/component";
import { useAppTheme } from "@src/contexts/ThemeContext";
import { getExtensionUrl } from "@src/utils/getExtensionUrl";

interface CursorButtonProps {
    onClick: () => void;
    color?: "white" | "yellow" | "purple";
    size?: "small" | "medium" | "large";
}

export function CursorButton({
    onClick,
    color = "white",
    size = "medium",
}: CursorButtonProps) {
    const { theme } = useAppTheme();

    const adjustedTheme = theme === "light" ? "white" : theme;

    const fileName = `cursor-${size}-${color}.png`;

    return (
        <BaseButton
            onClick={onClick}
            color={adjustedTheme}
            aria-label={`커서 변경 버튼: ${size}, ${color}`}
        >
            <img
                src={getExtensionUrl(fileName)}
                alt={`커서: ${size}, ${color}`}
                className="group-hover:opacity-20"
            />
        </BaseButton>
    );
}
