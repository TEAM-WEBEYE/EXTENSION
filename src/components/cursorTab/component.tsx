import { useTheme } from "@src/contexts/ThemeContext";
import React from "react";
import { CursorButton } from "../cursorButton";

type CursorSize = "small" | "medium" | "large";
type CursorColor = "white" | "black" | "yellow" | "purple" | "mint" | "pink";

const CURSOR_SIZES: CursorSize[] = ["small", "medium", "large"];
const CURSOR_COLORS: CursorColor[] = [
    "white",
    "black",
    "yellow",
    "purple",
    "mint",
    "pink",
];

export function CursorTab() {
    const { theme, fontClasses } = useTheme();
    const { setCursorSize, setCursorTheme, cursorSize, cursorTheme } =
        useTheme();

    const isDarkMode = theme === "dark";

    return (
        <div
            className={`font-koddi p-[18px] rounded-[20px] shadow-[0px_0px_4px_0px_rgba(0,0,0,0.25)] ${
                isDarkMode
                    ? `bg-grayscale-900 text-grayscale-100`
                    : `bg-grayscale-100 text-grayscale-900`
            }`}
        >
            <section className="mb-[26px]">
                <h2 className={fontClasses.fontCommon}>커서 크기 바꾸기</h2>
                <div className="flex gap-5 p-2">
                    {CURSOR_SIZES.map((size) => (
                        <CursorButton
                            key={size}
                            onClick={() => setCursorSize(size)}
                            size={size}
                            isSelected={cursorSize === size}
                        />
                    ))}
                </div>
            </section>

            <section>
                <h2 className={fontClasses.fontCommon}>커서 색상 바꾸기</h2>
                <div className="grid grid-cols-3 gap-5 p-2">
                    {CURSOR_COLORS.map((color) => (
                        <CursorButton
                            key={color}
                            onClick={() => setCursorTheme(color)}
                            color={color}
                            isSelected={cursorTheme === color}
                        />
                    ))}
                </div>
            </section>
        </div>
    );
}
