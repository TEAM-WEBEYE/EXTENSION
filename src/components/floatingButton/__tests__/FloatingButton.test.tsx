import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { FloatingButton } from "../component";
import { getExtensionUrl } from "@src/utils/getExtensionUrl";

jest.mock("@src/utils/getExtensionUrl", () => ({
    getExtensionUrl: jest
        .fn()
        .mockImplementation((path) => `/mocked-url/${path}`),
}));

describe("FloatingButton", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("스냅샷 테스트", () => {
        const { asFragment } = render(<FloatingButton onClick={() => {}} />);

        expect(asFragment()).toMatchSnapshot();
    });

    it("버튼이 올바르게 렌더링된다", () => {
        const handleClick = jest.fn();
        render(<FloatingButton onClick={handleClick} />);

        const button = screen.getByRole("button", { name: /voim/i });
        expect(button).toBeInTheDocument();
    });

    it("버튼 이미지가 올바르게 렌더링된다", () => {
        render(<FloatingButton onClick={() => {}} />);

        const image = screen.getByAltText("VOIM 익스텐션");
        expect(image).toBeInTheDocument();
        expect(getExtensionUrl).toHaveBeenCalledWith("icon.png");
        expect(image).toHaveAttribute("src", "/mocked-url/icon.png");
    });

    it("클릭 시 onClick 함수가 호출된다", () => {
        const handleClick = jest.fn();
        render(<FloatingButton onClick={handleClick} />);

        const button = screen.getByRole("button");
        fireEvent.click(button);

        expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it("Enter 키를 누를 때 onClick 함수가 호출된다", () => {
        const handleClick = jest.fn();
        render(<FloatingButton onClick={handleClick} />);

        const button = screen.getByRole("button");
        fireEvent.keyDown(button, { key: "Enter" });

        expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it("스페이스바를 누를 때 onClick 함수가 호출된다", () => {
        const handleClick = jest.fn();
        render(<FloatingButton onClick={handleClick} />);

        const button = screen.getByRole("button");
        fireEvent.keyDown(button, { key: " " });

        expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it("다른 키를 누를 때는 onClick 함수가 호출되지 않는다", () => {
        const handleClick = jest.fn();
        render(<FloatingButton onClick={handleClick} />);

        const button = screen.getByRole("button");
        fireEvent.keyDown(button, { key: "Tab" });

        expect(handleClick).not.toHaveBeenCalled();
    });

    it("올바른 ARIA 속성을 가지고 있다", () => {
        render(<FloatingButton onClick={() => {}} />);

        const button = screen.getByRole("button");
        expect(button).toHaveAttribute("aria-haspopup", "menu");
        expect(button).toHaveAttribute("aria-controls", "menubar");
        expect(button).toHaveAttribute("aria-expanded", "false");
        expect(button).toHaveAttribute("aria-label", "VOIM");
        expect(button).toHaveAttribute("tabIndex", "0");
    });

    it("클릭 후 aria-expanded 속성이 변경된다", () => {
        render(<FloatingButton onClick={() => {}} />);

        const button = screen.getByRole("button");

        expect(button).toHaveAttribute("aria-expanded", "false");

        fireEvent.click(button);
        expect(button).toHaveAttribute("aria-expanded", "true");

        fireEvent.click(button);
        expect(button).toHaveAttribute("aria-expanded", "false");
    });

    it("이미지에 aria-hidden 속성이 있다", () => {
        render(<FloatingButton onClick={() => {}} />);

        const image = screen.getByAltText("VOIM 익스텐션");
        expect(image).toHaveAttribute("aria-hidden", "true");
    });
});
