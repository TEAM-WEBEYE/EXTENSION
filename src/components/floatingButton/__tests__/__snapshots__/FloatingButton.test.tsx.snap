// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatingButton 스냅샷 테스트 1`] = `
<DocumentFragment>
  <div
    aria-controls="menubar"
    aria-expanded="false"
    aria-haspopup="menu"
    aria-label="VOIM"
    class="css-190eaou"
    role="button"
    tabindex="0"
  >
    <img
      alt="VOIM 익스텐션"
      aria-hidden="true"
      class="css-mn1htw"
      src="/mocked-url/icon.png"
    />
  </div>
</DocumentFragment>
`;
